monitors/test/ContractMonitor.js

```diff
@@ -0,0 +1,295 @@
+      sponsor1,
+      { rawValue: toWei("99999") },
+      { rawValue: toWei("100") },
+      { from: liquidator }
+    );
+
+    // Update the eventClient and check it has the liquidation event stored correctly
+    await eventClient._update();
+
+    // Check for liquidation events
+    await contractMonitor.checkForNewLiquidations(() => toWei("1"));
+
+    // Ensure that the spy correctly captured the liquidation events key information.
+    // Should contain etherscan addresses for the liquidator, sponsor and transaction
+    assert.isTrue(lastSpyLogIncludes(spy, `https://etherscan.io/address/${liquidator}`));
+    assert.isTrue(lastSpyLogIncludes(spy, "(Monitored liquidator bot)")); // The address that initiated the liquidation is a monitored address
+    assert.isTrue(lastSpyLogIncludes(spy, `https://etherscan.io/address/${sponsor1}`));
+    assert.isTrue(lastSpyLogIncludes(spy, `https://etherscan.io/tx/${txObject1.tx}`));
+
+    // should contain the correct position information. Collateralization ratio for sponsor with 150 collateral and 50
+    // debt with a price feed of 1 should give 150/(50 * 1) = 300%
+    assert.isTrue(lastSpyLogIncludes(spy, "300.00%")); // expected collateralization ratio of 300%
+    assert.isTrue(lastSpyLogIncludes(spy, "150.00")); // liquidated collateral amount of 150
+
+    // Liquidate another position and ensure the Contract monitor emits the correct params
+    const txObject2 = await emp.createLiquidation(
+      sponsor2,
+      { rawValue: toWei("99999") },
+      { rawValue: toWei("100") },
+      { from: sponsor1 } // not the monitored liquidator address
+    );
+
+    await eventClient._update();
+
+    // check for new liquidations and check the winston messages sent to the spy
+    await contractMonitor.checkForNewLiquidations(() => toWei("1"));
+    assert.isTrue(lastSpyLogIncludes(spy, `https://etherscan.io/address/${sponsor1}`)); // liquidator in txObject2
+    assert.isFalse(lastSpyLogIncludes(spy, "(Monitored liquidator bot)")); // not called from a monitored address
+    assert.isTrue(lastSpyLogIncludes(spy, `https://etherscan.io/address/${sponsor2}`)); // token sponsor
+    assert.isTrue(lastSpyLogIncludes(spy, `https://etherscan.io/tx/${txObject2.tx}`));
+    assert.isTrue(lastSpyLogIncludes(spy, "388.88%")); // expected collateralization ratio: 175 / (45 * 1) = 388.88%
+    assert.isTrue(lastSpyLogIncludes(spy, "175.00")); // liquidated collateral: 175
+  });
+  it("Winston correctly emits dispute events", async function() {
+    // Create liquidation to dispute.
+    await emp.createLiquidation(
+      sponsor1,
+      { rawValue: toWei("99999") },
+      { rawValue: toWei("100") },
+      { from: liquidator }
+    );
+
+    const txObject1 = await emp.dispute("0", sponsor1, {
+      from: disputer
+    });
+
+    // Update the eventClient and check it has the dispute event stored correctly
+    await eventClient.clearState();
+    await eventClient._update();
```

Comment:
Outside the scope of this PR, but I think you should add a wrapper `update()` method that is intended to be called publicly, similar to Liquidator/Disputer/EMPClient. I think the leading underscore ("_") implies that this method is meant to be called internally.